### PR TITLE
A few changes / fixes to mirror circuits

### LINKED ISSUE #905: Support any two-qubit gate in mirror circuits
Right now mirror circuits only have CNOTs. We want to support other two-qubit gates motivated by hardware providers with other gate sets, e.g., CZ instead of CNOT.

## Desired behavior

Add a `two_qubit_gate` parameter to `generate_mirror_circuit`. This should be front-end agnostic (e.g., not a `cirq.Gate` or `pyquil.Gate`).

```python
def generate_mirror_circuit(
    nlayers: int,
    two_qubit_gate_prob: float,
    two_qubit_gate: str,
    connectivity_graph: nx.Graph,
    seed: Optional[int] = None,
    return_type: Optional[str] = None,
) -> QPROGRAM:
```

Every two-qubit gate in the circuit is then an instance of the provided argument, e.g. CZ instead of CNOT, and the rest of the behavior of mirror circuits is the same.

### LINKED ISSUE #918: Qubits not in the `connectivity_graph` can be included in a mirror circuit
MWE

```python
import networkx as nx
from mitiq.benchmarks import generate_mirror_circuit

computer = nx.Graph({3: (4,)})
circuit = generate_mirror_circuit(1, 1, computer, 1)
print(circuit)
```

```text
0: ────────────────────X───────I───────I────────────────────

1: ────────────────────X───────X───────Y────────────────────

3: ───Y───────X^-0.5───────@───────@───────X^0.5───Y────────
                           │       │
4: ───Y^0.5───X^-0.5───────X───────X───────X^0.5───Y^-0.5───

```

Only qubits `3` and `4` should be present in the circuit - these are the only ones in the `computer` - but qubits `0` and `1` are also in the circuit.

### PR DESCRIPTION
**Pull request by rmlarose**
Description
-----------

This allows for any two-qubit gate in mirror circuits and fixes #905, and also fixes #918. Additionally, change `generate_mirror_circuit` to also return the bitstring that should be sampled by the circuit for convenience. 

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-types` (from the root directory of the repository) and fix any [mypy](https://mypy.readthedocs.io/en/stable/) errors.

    2. Run `make check-style` and fix any [flake8](http://flake8.pycqa.org) errors.

    3. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html) autoformatter.

  For more information, check the [Mitiq style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

**Comment by rmlarose** [comment]
PTAL @DSamuel1!

**Comment by DSamuel1** [comment]
Looks good to me @rmlarose . Thanks for the fixes!

File: mitiq/benchmarks/randomized_benchmarking.py [hunk 1]
```diff
@@ -45,8 +45,9 @@ def generate_rb_circuits(
             random circuits. This is proportional to the depth per circuit.
         trials: The number of random circuits at each num_cfd.
         return_type: String which specifies the type of the
-            returned circuits. See the keys of ``mitiq.QPROGRAM`` for options.
-            If ``None``, the returned circuits are type ``cirq.Circuit``.
+            returned circuits. See the keys of
+            ``mitiq.SUPPORTED_PROGRAM_TYPES`` for options. If ``None``, the
+            returned circuits are ``cirq.Circuit``s.
```
Comment (rmlarose) [inline mitiq/benchmarks/randomized_benchmarking.py:50]:
```suggestion
            returned circuits have type ``cirq.Circuit``.
```

**Comment by nathanshammah** [comment]
Great fix! Maybe if we move to release Mitiq 0.10, then this also helps with the notebooks to allow doing 
```
pip install mitiq
```
in the first block. We can then make another release for the notebook if we need more time.

**Pull request merged**